### ISSUE TITLE
depends_on parameter on data "aks_nodes_rg" force read

### ISSUE THREAD

**Issue by Ston9** [claranet/terraform-azurerm-aks#1]
I just deployed an AKS cluster using the module, but I continusly have this change : 

```
  # module.aks.data.azurerm_resource_group.aks_nodes_rg will be read during apply
  # (config refers to values not yet known)
 <= data "azurerm_resource_group" "aks_nodes_rg"  {
      + id       = (known after apply)
      + location = (known after apply)
      + name     = "my-nodes-rg"
      + tags     = (known after apply)

      + timeouts {
          + read = (known after apply)
        }
    }

  # module.aks.azurerm_role_assignment.aks_user_assigned must be replaced
-/+ resource "azurerm_role_assignment" "aks_user_assigned" {
      ~ id                               = "/subscriptions/xxx/resourceGroups/my-nodes-rg/providers/Microsoft.Authorization/roleAssignments/d3968cec-5ba9-5bba-f3a1-be1e164dd498" -> (known after apply)
      ~ name                             = "d3968cec-5ba9-5bba-f3a1-be1e164dd498" -> (known after apply)
        principal_id                     = "xxx"
      ~ principal_type                   = "ServicePrincipal" -> (known after apply)
      ~ role_definition_id               = "/subscriptions/xxx/providers/Microsoft.Authorization/roleDefinitions/b24988ac-6180-42a0-ab88-20f7382dd24c" -> (known after apply)
        role_definition_name             = "Contributor"
      ~ scope                            = "/subscriptions/xxx/resourceGroups/my-nodes-rg" -> (known after apply) # forces replacement
      + skip_service_principal_aad_check = (known after apply)
    }
```

Looking at this issue : https://github.com/hashicorp/terraform/issues/11806 , it seems that the change is related to the [depends_on] parameter on the `data resource` : 

```
data "azurerm_resource_group" "aks_nodes_rg" {
  depends_on = [azurerm_kubernetes_cluster.aks]
  name       = azurerm_kubernetes_cluster.aks.node_resource_group
}
```

Removing the line fix the issue, but will probably cause an other issue on the first apply of this module.

**Comment by jmapro** [comment]
May be we can remove this datasource. The Kubernetes nodes RG name is predictable and if it's a custom name we have the name in a variable.
I'll check that when I'll have some time or maybe @rossifumax will have some time to check that ;)

**Comment by jmapro** [comment]
Fixed in v4.0.0/v3.3.0
